### DISCUSSION TITLE
App plugin example to be used with grafana-toolkit cli

### DISCUSSION THREAD

**Discussion by jkhalack** [grafana/grafana#38350] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Please add an app plugin example that would work out of the box for Grafana 6.2+

**Why is this needed**:
Currently there is no complete app plugin example that shows how to set up an app plugin with all the dependencies.

- The [example-app](https://github.com/grafana/example-app) link on the [Developer Guide](https://grafana.com/docs/plugins/developing/development/#examples) does not exist any more (and the legacy apps created from that example seem to be incompatible with 6.2+ Grafana)
- The [example-app](https://github.com/grafana/grafana/tree/master/public/app/plugins/app/example-app) folder mentioned on [Developing App Plugins](https://grafana.com/docs/plugins/developing/apps/#develop-your-own-app) page does not have a `package.json` file (and if I try to reuse the dependencies from simple-panel-plugin repo, the build is failing at prettier check - so I must be trying to use incorrect dependencies).
- At the same time, `grafana-toolkit plugin:create` does not have an `app` option.

**Comment by pbaumard** [comment]
[influxdb-flux-datasource](https://github.com/grafana/influxdb-flux-datasource) would be a good plugin sample as it uses grafana-toolkit, but no tests are found by yarn test ([issue #64](https://github.com/grafana/influxdb-flux-datasource/issues/64)).

**Comment by jkhalack** [comment]
> [influxdb-flux-datasource](https://github.com/grafana/influxdb-flux-datasource) would be a good plugin sample as it uses grafana-toolkit, but no tests are found by yarn test ([issue #64](https://github.com/grafana/influxdb-flux-datasource/issues/64)).

That is a `datasource`, not an `app` plugin.

**Comment by gopalakrishnan-subramani** [comment]
The example for app is not shown up in my grafana portal ,where as I could build custom Datasource and custom panels using React, but the app type plugins is not seems to be working, it is not listed in the grafana portal. i tried building sensu app on my system, I could see the sensu app loaded rightly. 
Is below React and Custom app is tested, 
https://github.com/grafana/grafana/tree/master/public/app/plugins/app/example-app


anyone give working fork of https://github.com/grafana/example-app please as we got struck on this issue for a while.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We have several plugin examples listed here https://grafana.com/developers/plugin-tools/plugin-examples/ 

If anyone has questions about them, please ask  the [community forum](https://community.grafana.com/). Thank you!
